src/Microsoft.VisualStudio.Composition/Configuration/CachedCatalog.cs

```diff
@@ -125,23 +129,29 @@ private void Write(ComposablePartDefinition partDefinition)
                 }
             }
 
+            private Func<ImportDefinitionBinding> ReadImportDefinitionBindingDelegate => this.readImportDefinitionBinding ??= this.ReadImportDefinitionBinding;
+
+            private Func<ExportDefinition> ReadExportDefinitionDelegate => this.readExportDefinition ??= this.ReadExportDefinition;
+
+            private Func<IImportSatisfiabilityConstraint?> ReadIImportSatisfiabilityConstraintDelegate => this.readIImportSatisfiabilityConstraint ??= this.ReadIImportSatisfiabilityConstraint;
+
```

Comment:
Was there some particular reason you lazily initialize these in properties rather than initialize the fields in the ctor and use them directly?

Reply:
I'm happy to change it if you'd prefer.  I wasn't sure about whether there other use cases where all of these may not be used,  in which case the proactive allocation could be a deoptimization. Of that's not a concern,  I prefer the simplicity and efficiency of just allocating them all up front into readonly fields.

Reply:
I'm sure we'll end up initializing all of them, yes. Do you have to adjust this to initialize into `readonly` fields?

Reply:
I'll push another commit.